### ISSUE TITLE
Hard to use to those with no rust clue - or maybe doesn't work with recent nightly rust builds

### ISSUE THREAD

**Issue by djbclark** [welbornprod/phonewords#3]
(a) Mac OS X, installed rustc 1.10.0-nightly (8da2bcac5 2016-04-28) via curl method described on rust website's download page. IMHO it's always useful for a project to list the latest specific known-working compiler versions and dep versions; I'm guessing my problems may just be caused by rust being a fast-moving target that no longer works with the phonewords code even if one does everything right.  

(b) Doc said clippy was a dep so go and tried to install it, but its page didn't have any build or install instructions, eventually found and followed instructions for what looks like another version of the project, https://travis-ci.org/arcnmx/cargo-clippy but not sure those are at all right. 

(c) Also didn't see any build/install instructions for your project, but tried "cargo build" like in the travis-ci instructions, but just got a bunch of errors. Including log (note to others: to get this had to "export TERM=dumb" and then run "script" (/run stuff / ctrl-d) since cargo makes usual >file 2>&1 not work and also uses color by default which translates into a bunch of control characters in the typescript file.) as an attachment, zipped because it is 47 MB (!) uncompressed. 
[phonewords-build-1.txt.zip](https://github.com/welbornprod/phonewords/files/245598/phonewords-build-1.txt.zip)

(d) It looks like the "cargo build" in your project automatically downloaded/built/installed (possibly a different version of) clippy. Would be good to mention this in the README.md, e.g "clippy is a dependency (which is automatically installed during the phonewords build process so there is no reason to follow this link to its github page unless you really want to know what it is)"

(e) Since everyone expects to just type "make" to build and install stuff, it would be nice to have a (likely <10 line) Makefile that wraps cargo test/build/whatever (assuming cargo build is in fact what should be used :-) or just build instructions (exactly what to type) somewhere.

Your project looks super-useful and I hope to be able to get it working. I'm doing some installation art and searching for a phone number for a tree - or specifically the lighting for a tree, people will be able to call it and it'll respond to their voice in rave-party-lights fashion. I managed to get around 5 million phone numbers to search for via voip.ms API, and it turns out my initial plan of putting those in postgresql and doing everything via SQL would require an order of magnitude more computing resources than I have, at least.

In case anyone happens upon this and wants to get a near-complete list of US phone numbers available via voip.ms ... note that it takes hours to run.

```
# Depends on the "resty" and "json" command line tools.

source /usr/local/bin/resty
resty https://voip.ms/api/v1

# Note need to escape "+" and possibly other special chars in VUSER.
# You do not need to escape the "@" character.
VUSER="INSERT USER EMAIL ADDRESS HERE"
VPASS="INSERT API-SPECIFIC PASSWORD HERE"

GET /rest.php -d "api_username=${VUSER}" -d "api_password=${VPASS}" -d "method=getStates" -G | json states | json -a state > voipms-states.txt

while read STATE; do GET /rest.php -d "api_username=${VUSER}" -d "api_password=${VPASS}" -d "method=getRateCentersUSA" -d "state=${STATE}" -G | json ratecenters | json -a ratecenter >> voipms-ratecenters-${STATE}.txt; sleep 1; done < voipms-states.txt

for FILE in voipms-ratecenters-[A-Z][A-Z].txt; do
STATE="$(echo $FILE | sed 's/voipms\-ratecenters\-//g' | sed 's/\.txt//g')"
while read RATECENTER; do GET /rest.php -d "api_username=${VUSER}" -d "api_password=${VPASS}" -d "method=getDIDsUSA" -d "state=${STATE}" -d "ratecenter=${RATECENTER}" -G | json dids | json -a did >> voipms-dids-${STATE}.txt; sleep 1; done < "${FILE}"
done
done

cat voipms-dids-*.txt >> phonenums.tmp
awk 'length($0)==10' phonenums.tmp > phonenums.txt
```

**Comment by cjwelborn** [comment]
Thanks for the detailed report. I haven't been paying very much attention to this little project, and the dependencies were stale. Rust was changing a lot when I originally wrote this. I was trying to get faster results than my [multi-process Python version](https://github.com/welbornprod/wp_site/blob/dev/apps/phonewords/phone_words.py) that backs my [little web app](https://welbornprod.com/apps/phonewords) with the same functionality. This single-threaded Rust version is about the same, sometimes slower, than the multi-process Python version. I'm sure it could be faster though.

Those errors were mostly coming from `clippy` and `itertools`, but there were a couple small issues in the `phonewords` code itself. You probably got the latest versions of those (because I didn't specify a version in my `Cargo.toml`), and the `phonewords` code was still stuck in the past. Anyway, I've updated it to run with the latest `1.10.0-nightly (8da2bcac5 2016-04-28)`, `clippy 0.0.64`, and `itertools 0.4.13`. 

I put this up here for other people to benefit from. What you get when you do `cargo run --release` is a small command-line utility with barely any features. You would probably need to rip out the `check_number()` function and friends, and then adapt it for your own use case. I'm still no pro when it comes to Rust, but if you need any help I'm here.

I think I answered (_a_) (dependencies, moving targets, etc.). For (_b_), (_c_), and (_d_), you probably already guessed this, but `cargo run` or `cargo build` will handle the dependencies for you. For (_d_), I made a Makefile.  There's not much in it, but maybe it will point people in the right direction. Simply running `make` should build a working executable in `./target/release`.

You also mentioned a README, and I agree that any project should have a decent one. That's my fault. I guess I was assuming that only [rustaceans](http://www.rustaceans.org/) would be viewing this project, and I was wrong. I've fixed it up with some build instructions and some notes.

Thanks again for the issue, I'm glad to see some interest. If you post anything about your project (videos, pictures, blogs, etc.) I would love to see it. I dabble with micro-controllers sometimes and I like to see what other people have made.

**TLDR;** It should be fixed, `git pull && make`.
